### PR TITLE
DRY principle, examples with WET and DRY code

### PR DESCRIPTION
**Pull request by Pragatheesh**
I have added examples with WET code and then the DRY code with the initial code cleaned up.

PS: Also added a gitignore, with the initial line it just to ignore my IDE specific files

### CODE REVIEW

File: 5-dry/1-before.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    "startDate" => '2019-10-01',
+    "endDate" => '2019-10-05',
+    "total" => '10000000'
+];
+
+// Print the sales data in the console
+(new ReportProcessor())->show($salesData);
+
+// Extract the sales data into a file
+(new ReportProcessor())->show($salesData);
```
Comment (gayansanjeewa) [inline 5-dry/1-before.php]:
```suggestion
(new ReportProcessor())->exportToFile($salesData);
```

File: 5-dry/2-after.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    "startDate" => '2019-10-01',
+    "endDate" => '2019-10-05',
+    "total" => '10000000'
+];
+
+// Print the sales data in the console
+(new ReportProcessor())->show($salesData);
+
+// Extract the sales data into a file
+(new ReportProcessor())->show($salesData);
```
Comment (gayansanjeewa) [inline 5-dry/2-after.php]:
```suggestion
(new ReportProcessor())->saveToFile($salesData);
```

File: 5-dry/2-after.php [hunk 3]
```diff
@@ -0,0 +1,71 @@
+<?php
+
+// DRY - Don't Repeat Yourself
+class ReportProcessor
+{
+    /**
+     * @param array $salesData
+     */
+    public function show(array $salesData)
```
Comment (gayansanjeewa) [inline 5-dry/2-after.php]: I'm wondering whether `log()` would be a good name instead?

File: 5-dry/2-after.php [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    {
+        echo $this->createReport($salesData);
+    }
+
+    /**
+     * @param array $salesData
+     */
+    public function saveToFile(array $salesData)
```
Comment (gayansanjeewa) [inline 5-dry/2-after.php]: what about `export()` ?

**Review by gayansanjeewa — COMMENTED** [review]
Other than these it's okay for me

**Comment by Pragatheesh** [comment]
@gayansanjeewa the comments are addressed, please have a look at this again

**Pull request merged**